packages/paima-sdk/paima-crypto/src/mina.ts

```diff
@@ -0,0 +1,42 @@
+import { doLog } from '@paima/utils';
+import type { IVerify } from './IVerify.js';
+
+export class MinaCrypto implements IVerify {
+  verifyAddress = async (address: string): Promise<boolean> => {
+    // base58 alphabet
+    return await Promise.resolve(
+      /^[1|2|3|4|5|6|7|8|9|A|B|C|D|E|F|G|H|J|K|L|M|N|P|Q|R|S|T|U|V|W|X|Y|Z|a|b|c|d|e|f|g|h|i|j|k|m|n|o|p|q|r|s|t|u|v|w|x|y|z]/.test(
```

Comment:
This is only checking the first letter in the string. Also, does Mina use regular base58 or checked base58 (with a checksum)?

Probably it's easier to check the address validity with regular JS instead of regex (esp. if it's using the checksum variant)

Reply:
good catch... It's checksummed I think: https://github.com/o1-labs/o1js/blob/3b3bd1020a032b9a18d4d97ca445ca4c50b93cfd/src/lib/util/base58.ts#L36

But I'm not sure what can I use to access that function. I could add the `o1js` dependency I guess, and use `PublicKey.fromBech58` or something like that I guess? Because I think that module is not exposed.

Reply:
I assume their implementation is just `base58Check` and there are npm packages that provide just this functionality (not sure why o1js inlined it into their code instead of just using one of these packages)